launcher/ui/MainWindow.cpp

```diff
@@ -2050,6 +2063,73 @@ void MainWindow::on_actionKillInstance_triggered()
     }
 }
 
+void MainWindow::on_actionCreateInstanceShortcut_triggered()
+{
+    if (m_selectedInstance)
+    {
+        auto desktopPath = FS::getDesktopDir();
+        if (desktopPath.isEmpty()) {
+            // TODO come up with an alternative solution (open "save file" dialog)
+            QMessageBox::critical(this, tr("Create instance shortcut"), tr("Couldn't find desktop?!"));
+            return;
+        }
+
+#if defined(Q_OS_MACOS)
+        // macOSX
+        // TODO actually write this path
+        QMessageBox::critical(this, tr("Create instance shortcut"), tr("Not supported on macOSX yet!"));
+#else
+        auto icon = APPLICATION->icons()->icon(m_selectedInstance->iconKey());
+
+        QString iconPath;
+        bool iconGenerated = false;
+
+#if defined(Q_OS_WIN)
+        iconPath = FS::PathCombine(m_selectedInstance->instanceRoot(), "icon.ico");
+
```

Comment:
Having nested `#if` blocks might be confusing. Consider ending the macOS check with a return instead.

i.e.
```cpp
#ifdef Q_OS_MACOS
    // macOS
    // TODO actually write this path
    QMessageBox::critical(this, tr("Create instance shortcut"), tr("Creating shortcuts is not supported on macOS yet."));
    return;
#endif
```